### PR TITLE
golangci-lint: Update to v1.51.2

### PR DESCRIPTION
**Pull request by mhofstetter**
Updating golangci-lint to latest version improves performance when executing locally.

In addition, failing lint issues in LBIPAM have been fixed.

### CODE REVIEW

**Review by jibi — APPROVED** [review]

**Review by brlbil — APPROVED** [review]

File: operator/pkg/lbipam/lbipam.go [hunk 1]
```diff
@@ -923,8 +920,6 @@ func (ipam *LBIPAM) findRangeOfIP(sv *ServiceView, ip net.IP) (lbRange *LBRange,
 			continue
 		}
 
-		foundPool = true
```
Comment (tklauser) [inline operator/pkg/lbipam/lbipam.go]: I'm not sure whether the current code is correct at all. It seems we never return `foundPool=true` in any case. Maybe the final `return` in this func should be `return nil, foundPool, nil` and we keep this assignment here?

/cc @dylandreimerink who added this code

File: Makefile.defs [hunk 2]
```diff
@@ -87,8 +87,8 @@ ifneq ($(V),0)
 endif
 endif
 
-GOLANGCILINT_WANT_VERSION = 1.51.0
-GOLANGCILINT_IMAGE_SHA = sha256:06c7e2e557603b122aaa7db5568b6328b980c6746ba7eb9652cf4ca4359e42d6
+GOLANGCILINT_WANT_VERSION = 1.51.2
+GOLANGCILINT_IMAGE_SHA = sha256:b0464d4d425f95f73a3443496c68be5fc3b5c456d2bd379f0d29ebaf7eeab657
```
Comment (kaworu) [inline Makefile.defs]: I'm getting another hash

```
% docker inspect --format='{{index .RepoDigests 0}}' docker.io/golangci/golangci-lint:v1.51.2
golangci/golangci-lint@sha256:be42f5df32011553183947c2b2502718bde485f566e15bbd2541f38a077e4d29
```

File: operator/pkg/lbipam/lbipam.go (hunk 1 above)
Comment (dylandreimerink) [reply operator/pkg/lbipam/lbipam.go]: It isn't. This PR https://github.com/cilium/cilium/pull/23297 was supposed to address it already, but I have not gotten it merged yet.

Cool to see that the new linter finds this issue as well. :+1:

Comment (dylandreimerink) [reply operator/pkg/lbipam/lbipam.go]: As to not block this PR, I would suggest to go with the fix I proposed in mine. Which is to simply change the final return to `return nil, foundPool, nil` like @tklauser suggests. I will rebase https://github.com/cilium/cilium/pull/23297 on master once this is in

Comment (mhofstetter) [reply operator/pkg/lbipam/lbipam.go]: thanks @dylandreimerink & @tklauser

changed this to use the suggested return statement

File: operator/pkg/lbipam/lbipam.go [hunk 3]
```diff
@@ -939,7 +936,7 @@ func (ipam *LBIPAM) findRangeOfIP(sv *ServiceView, ip net.IP) (lbRange *LBRange,
 		return r, false, nil
```
Comment (kaworu) [inline operator/pkg/lbipam/lbipam.go:936]: shouldn't we return true for `foundPool` here too?
```suggestion
		return r, foundPool, nil
```

Comment (mhofstetter) [reply operator/pkg/lbipam/lbipam.go:936]: might be more a question for @dylandreimerink - which should be discussed/solved in the corresponding PR? https://github.com/cilium/cilium/pull/23297

but i'm happy to fix it here as well if it's the expected behaviour.

**Review by dylandreimerink — APPROVED** [review]
LGTM!

**Review by tklauser — APPROVED** [review]
LGTM, thanks!

**Pull request merged**